appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,657 @@
+  """
+  Obtains useful information from process statistics and returns:
+  PRIVATE IP, MONIT NAME, UNIQUE MEMORY (MB), CPU (%) values.
+
+  Args:
+    raw_process_stats: A dist in which each key is an ip and value is a dict
+      of useful information.
+
+  Returns:
+    A list of process statistics headers.
+    A list of process statistics.
+  """
+  process_stats_headers = [
+    "PRIVATE IP", "MONIT NAME", "UNIQUE MEMORY (MB)", "CPU (%)"
+  ]
+
+  process_stats = []
+  for ip, node in raw_process_stats.iteritems():
+    stat = node.get("processes_stats")
+    for proc in stat:
+      memory_unique = int(proc["memory"]["unique"])
+      cpu_percent = float(proc["cpu"]["percent"])
+      if proc["children_num"] > 0:
+        child = proc["children_stats_sum"]
+        memory_unique += int(child["memory"]["unique"])
+        cpu_percent += float(child["cpu"]["percent"])
+      process_stats.append([
+        ip,
+        proc["monit_name"],
+        memory_unique / 1024 / 1024,  # row[2] from B to MB
+        cpu_percent
+      ])
+
+  return process_stats_headers, process_stats
+
+
+def get_summary_process_stats(raw_process_stats, raw_node_stats):
+  """
+  Obtains useful information from summary process statistics and returns:
+  SERVICE (ID), INSTANCES, UNIQUE MEMORY SUM (MB),
+  CPU SUM (%), CPU PER 1 PROCESS (%), CPU PER 1 CORE (%) values.
+
+  Args:
+    raw_process_stats: A dist in which each key is an ip and value is a dict
+      of useful information.
+
+  Returns:
+    A list of summary process statistics headers.
+    A list of summary process statistics.
+  """
+  cpu_count = sum(
+    int(node["cpu"]["count"])
+    for node in raw_node_stats.itervalues()
+  )
+
+  process_stats = []
+  process_list = []
+  unique_processes = set()
+
+  process_stats_headers = [
+    "SERVICE (ID)", "INSTANCES", "UNIQUE MEMORY SUM (MB)",
+    "CPU SUM (%)", "CPU PER 1 PROCESS (%)", "CPU PER 1 CORE (%)"
+  ]
+
+  for proc in raw_process_stats.itervalues():
+    process_list += proc["processes_stats"]
+
+  for proc in process_list:
+    name = proc["unified_service_name"]
+    app_id = proc["application_id"]
+    name_id = name + (" ({})".format(app_id) if app_id else "")
+    if name_id in unique_processes:  # if calculating with process has been made
+      continue
+
+    process_group = [
+      p for p in process_list
+      if p["unified_service_name"] == name
+      and p["application_id"] == app_id
+    ]
+
+    unique_processes.add(name_id)
+    unique_memory = 0
+    cpu_percent = 0
+
+    for p in process_group:
+      unique_memory += int(p["memory"]["unique"])
+      cpu_percent += float(p["cpu"]["percent"])
+      if p["children_num"] > 0:
+        child = p["children_stats_sum"]
+        unique_memory += int(child["memory"]["unique"])
+        cpu_percent += float(child["cpu"]["percent"])
+
+    avg_cpu_per_process = cpu_percent / len(process_group)
+    cpu_percent_cpu_count = cpu_percent / cpu_count
```

Comment:
Okay, but if I clarify that I show only AppScale processes?
I mean, the table name will be "APPSCALE PROCESS STATISTICS" instead of "PROCESS STATISTICS".